src/Workspaces/CSharp/Portable/Utilities/SpeculationAnalyzer.cs

```diff
@@ -528,6 +528,17 @@ protected override bool IsInvocableExpression(SyntaxNode node)
             return false;
         }
 
+        protected override bool IsMemberAccessExpression(SyntaxNode node)
```

Comment:
There are ISyntaxFactsService helpers for these. `IsSimpleMemberAccessExpression` [here](http://source.roslyn.io/#Microsoft.CodeAnalysis.Workspaces/LanguageServices/SyntaxFactsService/ISyntaxFactsService.cs,e87725746ba9b005,references) and friends.